internal/runbits/requirements/requirements.go

```diff
@@ -0,0 +1,334 @@
+package requirements
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	anaConsts "github.com/ActiveState/cli/internal/analytics/constants"
+	"github.com/ActiveState/cli/internal/captain"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/installation/storage"
+	"github.com/ActiveState/cli/internal/keypairs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/prompt"
+	"github.com/ActiveState/cli/internal/runbits"
+	medmodel "github.com/ActiveState/cli/pkg/platform/api/mediator/model"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/ActiveState/cli/pkg/projectfile"
+	"github.com/go-openapi/strfmt"
+	"github.com/thoas/go-funk"
+)
+
+type PackageVersion struct {
+	captain.NameVersion
+}
+
+func (pv *PackageVersion) Set(arg string) error {
+	err := pv.NameVersion.Set(arg)
+	if err != nil {
+		return locale.WrapInputError(err, "err_package_format", "The package and version provided is not formatting correctly, must be in the form of <package>@<version>")
+	}
+	return nil
+}
+
+type primeable interface {
+	primer.Outputer
+	primer.Prompter
+	primer.Projecter
+	primer.Auther
+	primer.Configurer
+	primer.Analyticer
+	primer.SvcModeler
+}
+
+type configurable interface {
+	keypairs.Configurable
+}
+
+const latestVersion = "latest"
+
+func ExecuteRequirementOperation(prime primeable, requirementName, requirementVersion string, operation model.Operation, nsType model.NamespaceType) (rerr error) {
```

Comment:
This seems like you're introducing a ton of new code.. but I'm guessing this is actually just mostly copying ExecutePackageOperation? Unless github is being weird again it does not appear that ExecutePackageOperation went anywhere though? Can you clarify what's happening?

Reply:
Whoops. Forgot to remove the `executePackageOperation` function. The new `ExecuteRequirementOperation` is just a modified version of the original package operation function.